palindrome_linked_list.cpp

```diff
@@ -0,0 +1,152 @@
+#define CATCH_CONFIG_MAIN
+#include "catch.hpp"
+#include <iostream>
+#include <stack>
+#include <string>
+
+using namespace std;
+
+// basic linked list class
+class linkedList {
```

Comment:
It's good practice to use PascalCasing for class names. Whichever method you prefer, make sure that your class names have different casing than local variable names.